substrate/frame/nomination-pools/src/weights.rs

```diff
@@ -675,59 +677,40 @@ impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
 			.saturating_add(T::DbWeight::get().writes(2_u64))
 	}
-	/// Storage: `NominationPools::PoolMembers` (r:1 w:0)
-	/// Proof: `NominationPools::PoolMembers` (`max_values`: None, `max_size`: Some(237), added: 2712, mode: `MaxEncodedLen`)
 	fn apply_slash() -> Weight {
 		// Proof Size summary in bytes:
-		//  Measured:  `694`
-		//  Estimated: `3702`
-		// Minimum execution time: 13_403_000 picoseconds.
-		Weight::from_parts(14_064_000, 3702)
-			.saturating_add(T::DbWeight::get().reads(1_u64))
+		//  Measured:  `0`
+		//  Estimated: `0`
+		// Minimum execution time: 2_846_000 picoseconds.
+		Weight::from_parts(3_058_000, 0)
 	}
-	/// Storage: `NominationPools::PoolMembers` (r:1 w:0)
-	/// Proof: `NominationPools::PoolMembers` (`max_values`: None, `max_size`: Some(237), added: 2712, mode: `MaxEncodedLen`)
 	fn apply_slash_fail() -> Weight {
 		// Proof Size summary in bytes:
-		//  Measured:  `732`
-		//  Estimated: `3702`
-		// Minimum execution time: 14_419_000 picoseconds.
-		Weight::from_parts(15_004_000, 3702)
-			.saturating_add(T::DbWeight::get().reads(1_u64))
+		//  Measured:  `0`
+		//  Estimated: `0`
+		// Minimum execution time: 2_858_000 picoseconds.
+		Weight::from_parts(3_010_000, 0)
 	}
 	fn pool_migrate() -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 759_000 picoseconds.
-		Weight::from_parts(819_000, 0)
+		// Minimum execution time: 2_719_000 picoseconds.
+		Weight::from_parts(2_828_000, 0)
 	}
-	/// Storage: `NominationPools::PoolMembers` (r:1 w:0)
-	/// Proof: `NominationPools::PoolMembers` (`max_values`: None, `max_size`: Some(237), added: 2712, mode: `MaxEncodedLen`)
-	/// Storage: `NominationPools::BondedPools` (r:1 w:0)
-	/// Proof: `NominationPools::BondedPools` (`max_values`: None, `max_size`: Some(254), added: 2729, mode: `MaxEncodedLen`)
-	/// Storage: `Staking::Bonded` (r:1 w:0)
-	/// Proof: `Staking::Bonded` (`max_values`: None, `max_size`: Some(72), added: 2547, mode: `MaxEncodedLen`)
-	/// Storage: `Staking::Ledger` (r:1 w:0)
-	/// Proof: `Staking::Ledger` (`max_values`: None, `max_size`: Some(1091), added: 3566, mode: `MaxEncodedLen`)
-	/// Storage: `NominationPools::SubPoolsStorage` (r:1 w:0)
-	/// Proof: `NominationPools::SubPoolsStorage` (`max_values`: None, `max_size`: Some(24382), added: 26857, mode: `MaxEncodedLen`)
-	/// Storage: `NominationPools::MinJoinBond` (r:1 w:0)
-	/// Proof: `NominationPools::MinJoinBond` (`max_values`: Some(1), `max_size`: Some(16), added: 511, mode: `MaxEncodedLen`)
 	fn migrate_delegation() -> Weight {
 		// Proof Size summary in bytes:
-		//  Measured:  `1648`
-		//  Estimated: `27847`
-		// Minimum execution time: 36_192_000 picoseconds.
-		Weight::from_parts(37_038_000, 27847)
-			.saturating_add(T::DbWeight::get().reads(6_u64))
+		//  Measured:  `0`
+		//  Estimated: `0`
+		// Minimum execution time: 2_776_000 picoseconds.
+		Weight::from_parts(2_987_000, 0)
```

Comment:
Also surprising that this operation doesn't read or write anything.
The caller is not related to the operation.